soh/soh/Enhancements/randomizer/3drando/location_access/locacc_dodongos_cavern.cpp

```diff
@@ -235,7 +235,7 @@ void AreaTable_Init_DodongosCavern() {
   areaTable[DODONGOS_CAVERN_BOSS_ROOM] = Area("Dodongos Cavern Boss Room", "Dodongos Cavern", DODONGOS_CAVERN, NO_DAY_NIGHT_CYCLE, {
                   //Events
                   EventAccess(&DodongosCavernClear, {[]{return DodongosCavernClear || (Here(DODONGOS_CAVERN_BOSS_ROOM, []{return HasExplosives || (CanUse(MEGATON_HAMMER) && CanShield);}) && (Bombs || GoronBracelet) && (IsAdult || Sticks || KokiriSword));},
-                                         /*Glitched*/[]{return Here(DODONGOS_CAVERN_BOSS_ROOM, []{return HasExplosives || (CanUse(MEGATON_HAMMER) && CanShield) || (GlitchBlueFireWall && BlueFire);}) && (HasExplosives || GoronBracelet) && (IsAdult || Sticks || KokiriSword || CanUse(MEGATON_HAMMER));}}),
+                                         /*Glitched*/[]{return Here(DODONGOS_CAVERN_BOSS_ROOM, []{return HasExplosives || (CanUse(MEGATON_HAMMER) && CanShield) || (GlitchBlueFireWall && BlueFire && HasBottle && BlueFireAccess);}) && (HasExplosives || GoronBracelet) && (IsAdult || Sticks || KokiriSword || CanUse(MEGATON_HAMMER));}}),
```

Comment:
Why is this neccesary when the conditionals you added here already exist in `BlueFire`?

Reply:
They are overriding blue fire arrows being in logic here, since breaking the floor over King Dodongo's room is not in logic with blue fire arrows in N64 rando. Actually it doesn't seem to be in logic with regular Blue Fire over there either, but the other pieces of this were already here in 3DS rando. Apparently the hitbox of that floor is difficult to hit with the ice arrows.

Reply:
That might be the intent, but that's not what's actually happening here. Then BlueFire in here should be removed.

Reply:
I initially did say that it was fine since N64 also didn't have Blue Fire Arrows as logically able to enter, but since they also don't have blue fire in logic at all for getting to KD I do want to revisit that statement. I'm inclined to say that it's OK if the hitbox for the floor is a bit tricky with Blue Fire Arrows since it would only be required in glitched logic anyway.